bin/publish-gem.sh

```diff
@@ -0,0 +1,8 @@
+mkdir -p ~/.gem
```

Comment:
Is it possible to use https://github.com/marketplace/actions/publish-gem instead?

There's an example in sorbet-rails 
https://github.com/chanzuckerberg/sorbet-rails/blob/master/.github/workflows/publish-gem.yml

Reply:
Yup, can do. I took this approach since it's what Spencer had done for redcord, but the publish-gem workflow seems simpler. Can definitely adjust in this PR; should I also update redcord/sorbet-coerce?